pkg/providers/tinkerbell/create.go

```diff
@@ -118,7 +118,14 @@ func (p *Provider) SetupAndValidateCreateCluster(ctx context.Context, clusterSpe
 		validator.Register(NewIPNotInUseAssertion(p.netClient))
 	}
 
-	writer := hardware.NewMachineCatalogueWriter(p.catalogue)
+	catalogueWriter := hardware.NewMachineCatalogueWriter(p.catalogue)
+
+	for _, mc := range p.machineConfigs {
+		p.templateBuilder.diskExtractor.RegisterHardwareSelector(mc.Spec.HardwareSelector)
```

Comment:
We shouldn't be reaching into other constructs and modifying their members (Law of Demeter). 

Its not a great solution, but for now can we expose a `RegisterHardwareSelector` on the `TemplateBuilder` so we can call `p.templateBuilder.RegisterHardwareSelector()`. This, at least, maintains a degree of separation that will compliment any `TemplateBuilder` dependency injection we refactor for in the future.